cmake/kokkos_enable_options.cmake

```diff
@@ -48,7 +48,6 @@ KOKKOS_ENABLE_OPTION(COMPILER_WARNINGS    OFF "Whether to print all compiler war
 KOKKOS_ENABLE_OPTION(PROFILING            ON  "Whether to create bindings for profiling tools")
 KOKKOS_ENABLE_OPTION(PROFILING_LOAD_PRINT OFF "Whether to print information about which profiling tools got loaded")
 KOKKOS_ENABLE_OPTION(AGGRESSIVE_VECTORIZATION OFF "Whether to aggressively vectorize loops")
-KOKKOS_ENABLE_OPTION(DEPRECATED_CODE          OFF "Whether to enable deprecated code")
```

Comment:
@masterleinad is it reasonable to keep the DEPRECATED_CODE macro available for future code deprecation (i.e. in anticipation of Kokkos 4.0)?

Reply:
I am open to keep it, or remove and reintroduce again when necessary.